website/docs/tutorials/07-testing/04-e2e.md

```diff
@@ -0,0 +1,62 @@
+# End-to-end tests
+
+There are three main types of testing software: unit tests, integration tests and end-to-end tests.
+
+Protostar testing framework provides an environment to write unit tests for Cairo smart contracts
+in Cairo language itself.
+With help of cheatcodes like [`deploy_contract`] and [`invoke`] it is also easy to write
+integration tests, deploying single or a few contracts and examining internal behaviour of such
+small systems.
+
+## Recommended approach
+
+In end-to-end testing, the goal is to examine interactions not only within a system of contracts
+deployed on the blockchain, but also to examine interactions triggered by external services,
+like web frontends, and observing effects happening within these services themselves.
+
+:::note
+As an example, an end-to-end test of a voting application could trigger a click on a _Vote_ button,
+causing a transaction being performed on the blockchain.
+The test then waits for the transaction to be accepted, and examines changes of observable state of
+voting contract, and the results gathered by the voting application.
```

Comment:
```suggestion
voting contract, e.g. the final result of the voting.
```

Reply:
I wanted to tell about changes in UI of the app here. I will apply the change, but I miss this part now...